scripts/seedUsersAndPosts.js

```diff
@@ -0,0 +1,519 @@
+import 'dotenv/config';
+import db from '../config/mongoCollections.js';
+import userData from '../data/users.js';
+import postData from '../data/posts.js';
+import { userSchema } from '../models/users.js';
+import { postSchema } from '../models/posts.js';
+
+const seedUsersAndPosts = async () => {
+    try {
+        const usersCollection = await db.users();
+        const postsCollection = await db.posts();
+        
+        // Check if already seeded
+        const userCount = await usersCollection.countDocuments();
+        const postCount = await postsCollection.countDocuments();
+        
+        if (userCount > 0 || postCount > 0) {
+            console.log(`Database already has ${userCount} users and ${postCount} posts`);
+            console.log('Skipping seed. Delete collections to re-seed.');
+            return;
+        }
+
+        console.log('Starting user and post seed...');
+
+        // Create test users with Marvel characters
+        const testUsers = [
+            {
+                email: 'nick.fury@shield.gov',
+                password: 'Password123!',
+                role: 'admin',
+                zipcode: '10001', // NYC - Midtown
+                profile: {
+                    username: 'director-fury',
+                    firstName: 'Nick',
+                    lastName: 'Fury',
+                    dob: new Date('1950-12-21'),
+                    bio: 'Director of S.H.I.E.L.D. Keeping an eye on everything. (Admin account)',
+                    profilePicture: null
```

Comment:
Hey so I still need to review, but I don’t want to forget about this:

`profilePicture` will be called `avatar` moving forward. Nothing else in regards to this is required from you, other than this name change.

```suggestion
                    avatar: null
```